addons/mail/wizard/mail_resend_message.py

```diff
@@ -90,13 +96,32 @@ def cancel_mail_action(self):
         return {'type': 'ir.actions.act_window_close'}
 
 
-class PartnerResend(models.TransientModel):
-    _name = 'mail.resend.partner'
-    _description = 'Partner with additional information for mail resend'
+class ContactResend(models.TransientModel):
+    _name = 'mail.resend.contact'
+    _description = 'Contact information for mail resend'
+
+    notification_id = fields.Many2one('mail.notification', string='Notification', ondelete='cascade', readonly=True)
+    partner_id = fields.Many2one('res.partner', related='notification_id.res_partner_id', string='Partner', ondelete='cascade')
+    name = fields.Char(related='partner_id.name', string='Recipient Name')
+    email = fields.Char(compute='_compute_email', inverse="_inverse_email", string='Email Address')
+
+    @api.depends('partner_id', 'notification_id')
+    def _compute_email(self):
+        for contact in self:
+            contact.email = contact.notification_id.unpartnered_email or contact.partner_id.email
+
+    def _inverse_email(self):
+        for contact in self:
+            if contact.partner_id:
+                contact.partner_id.email = email_normalize(contact.email)
+
+            elif contact.notification_id.unpartnered_email:
+                if contact.notification_id.notification_status in ('bounce', 'exception'):
+                    # This is fine because the mail has not been sent yet, so it will not be falsified
+                    contact.notification_id.sudo().unpartnered_email = email_normalize(contact.email)
+                else:
+                    raise AccessError(_("You may not modify the email of a notification that is not waiting to be re-sent"))
```

Comment:
check also other occurrences
```suggestion
                    raise AccessError _("You may not modify the email of a notification that is not waiting to be re-sent")
```

Reply:
Mm, I think that might actually be python2 syntax? Seems like it throws a syntax error if I don't use it like a constructor.

Reply:
ah yes nevermind, the parenthesis was around the assert in my mind